### PR TITLE
implement constant time key check

### PR DESCRIPTION
**Pull request by StarpTech**
- implement constant time key check,
- support arrays and sets as keys
- simplify module
- remove `'` in scripts to support windows
- update deps

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,8 +4,8 @@
   "description": "An authentication plugin for Fastify",
   "main": "plugin.js",
   "scripts": {
-    "test": "tap 'test/**/*.test.js'",
-    "test-ci": "tap --cov 'test/**/*.test.js'",
```
Comment (jsumners) [inline package.json]: These `'` need to be present to prevent shell expansion of the globs.

File: plugin.js [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
 
 const fp = require('fastify-plugin')
+const compare = require('secure-compare')
 
-const internals = {}
-internals.factory = function (options) {
+function factory (options) {
   const defaultOptions = {
-    keys: new Set(),
+    keys: [],
```
Comment (jsumners) [inline plugin.js]: Why is this no longer a `Set`? At the very least, the documentation needs to be updated to reflect this change.

Comment (StarpTech) [reply plugin.js]: no, it's compatible with `Set` and `arrays`. A set and array have different interfaces.

File: package.json (hunk 1 above)
Comment (StarpTech) [reply package.json]: done I'm using quotes

File: plugin.js (hunk 2 above)
Comment (StarpTech) [reply plugin.js]: @jsumners I updated the docs

File: Readme.md [hunk 3]
```diff
@@ -37,13 +37,15 @@ you to register the plugin within scoped paths. Therefore, you could have some
...
-When registering the plugin you must specify a configuration object that at
-least has a `keys` property set to an object that supports a `has(key)` method.
-It may also have a method `errorResponse(err)` and property `contentType`. If set,
-the `errorResponse(err)` method must synchronously return the content body to be
-sent to the client. If the content to be sent is anything other than
-`application/json`, then the `contentType` property must be set. The default
-config object is:
+When registering the plugin you must specify a configuration object:
+
+* keys: A `Set` or array with valid keys of type `string` (required)
```
Comment (jsumners) [inline Readme.md]: > `keys`:

File: Readme.md [hunk 4]
```diff
@@ -37,13 +37,15 @@ you to register the plugin within scoped paths. Therefore, you could have some
...
+* errorResponse(err): method must synchronously return the content body to be
```
Comment (jsumners) [inline Readme.md]: > `function errorResponse (err) {}`:

File: Readme.md [hunk 5]
```diff
@@ -37,13 +37,15 @@ you to register the plugin within scoped paths. Therefore, you could have some
...
+sent to the client (optional)
+* contentType: If the content to be sent is anything other than
```
Comment (jsumners) [inline Readme.md]: > `contentType`:

File: Readme.md [hunk 6]
```diff
@@ -37,13 +37,15 @@ you to register the plugin within scoped paths. Therefore, you could have some
...
+`application/json`, then the `contentType` property must be set (optional)
+
+The default config object is:
```
Comment (jsumners) [inline Readme.md]: "config" => "configuration"

**Comment by StarpTech** [comment]
I have no rights on npm

**Comment by jsumners** [comment]
For any "official" ecosystem package? Or just for this one? @mcollina or @delvedor can you check the permissions? I won't be able to issue a release until the weekend at the earliest.

**Comment by StarpTech** [comment]
Thanks

**Pull request merged**